code/game/centcomm_orders/centcomm_orders.dm

```diff
@@ -0,0 +1,386 @@
+	for(var/typepath in left_to_check)
+		var/worth_per_unit = unit_prices[typepath]
+		var/amount         = left_to_check[typepath]
+		toPay += amount * worth_per_unit
+		if(requested[typepath] != INFINITY)
+			requested[typepath] = max(0,requested[typepath] - left_to_check[typepath])
+		if(!(typepath in fulfilled))
+			fulfilled[typepath] = 0
+		fulfilled[typepath] += left_to_check[typepath]
+		left_to_check[typepath] = 0
+	. = ..()
+	Pay(.)
+
+///////////////////////////////
+
+/proc/create_random_orders(var/num_orders)//This one is used at roundstart to add a couple random orders immediately
+	var/list/choices = get_all_orders()
+	for(var/i = 1 to num_orders)
+		var/choice = pick_n_take(choices)
+		var/datum/centcomm_order/new_order = new choice
+		SSsupply_shuttle.add_centcomm_order(new_order)
+
+/proc/get_all_orders()
+	var/list/orders = list()
+	orders.Add(subtypesof(/datum/centcomm_order/per_unit/department/cargo))
+	orders.Add(subtypesof(/datum/centcomm_order/department/science))
+	orders.Add(subtypesof(/datum/centcomm_order/department/medical))
+	orders.Add(subtypesof(/datum/centcomm_order/department/engineering))
+	orders.Add(subtypesof(/datum/centcomm_order/department/civilian))
+	orders.Add(subtypesof(/datum/centcomm_order/per_unit/department/civilian))
+	return orders
+
+
+///////////////////////////////////////
+
+/proc/create_weighted_order()
+	var/list/active_with_role = get_dept_pop()
+
+	var/list/department_weights = list(
+		"Cargo" = 3,
+		"Service" = 5,
+		"Medical" = 5,
+		"Science" = 5,
+		"Engineering" = 5,
+		)
+
+	for(var/dept in department_weights)
+		if(active_with_role[dept] < 1)
+			department_weights[dept] = 1//departments with no employees are very unlikely to receive an order
+
+	var/list/order_exists = list(
+		"Cargo" = 0,
+		"Service" = 0,
+		"Medical" = 0,
+		"Science" = 0,
+		"Engineering" = 0,
+		)
+
+	for(var/datum/centcomm_order/O in SSsupply_shuttle.centcomm_orders)
+		if (istype(O, /datum/centcomm_order/per_unit/department/cargo))
+			order_exists["Cargo"] += 1
+		if (istype(O, /datum/centcomm_order/department/science))
+			order_exists["Science"] += 1
+		if (istype(O, /datum/centcomm_order/department/medical))
+			order_exists["Medical"] += 1
+		if (istype(O, /datum/centcomm_order/department/engineering))
+			order_exists["Engineering"] += 1
+		if (istype(O, /datum/centcomm_order/per_unit/department/civilian))
+			order_exists["Service"] += 1
+		if (istype(O, /datum/centcomm_order/department/civilian))
+			order_exists["Service"] += 1
+
+	for(var/dept in order_exists)
+		department_weights[dept] = max(1, department_weights[dept] - order_exists[dept])//the more active orders a department has, the less likely it'll get another one
+
+	var/chosen_dept = pick(
+		department_weights["Cargo"];"Cargo",
+		department_weights["Service"];"Service",
+		department_weights["Medical"];"Medical",
+		department_weights["Science"];"Science",
+		department_weights["Engineering"];"Engineering")
+
+	if (!chosen_dept)
```

Comment:
I don't think this will ever pass. `pick` runtimes if it tries to pick from an empty list.